extensions/resource-deployment/src/ui/model.ts

```diff
@@ -32,6 +33,33 @@ export class Model {
 		return value === undefined ? defaultValue : value === 'true';
 	}
 
+	/**
+	 * returns python code statements for setting variables starting with {@see NoteBookEnvironmentVariablePrefix} as python variables.
+	 * The prefix {@see NoteBookEnvironmentVariablePrefix} is removed and variable name changed to all lowercase to arrive at python variable name.
+	 * The statements returned are escaped for use in cell of a python notebook.
+	 */
+	public getCodeCellContentForNotebook(): string[] {
+		const regex = new RegExp(`^${NoteBookEnvironmentVariablePrefix}`);
+		const statements: string[] = Object.keys(this.propValueObject)
+			.filter(propertyName => propertyName.startsWith(NoteBookEnvironmentVariablePrefix))
+			.map(propertyName => {
+				const value = this.escapeForNotebookCodeCell(this.getStringValue(propertyName, ''));
+				const varName = propertyName.replace(regex, '').toLocaleLowerCase();
```

Comment:
This is only going to replace the first occurrence, is that intended?

Reply:
no. I will put the 'g' flag in regex.

Reply:
Yes the regex matches NoteBookEnvironmentVariablePrefix at the beginning of the string so only one replacement is expected.